compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -609,6 +610,178 @@ setMatmulVectorDistributionConfig(IREE::GPU::TargetAttr target,
                                                targetSubgroupSize, configDict);
 }
 
+static LogicalResult
+setAttentionVectorDistributionConfig(IREE::GPU::TargetAttr target,
+                                     mlir::FunctionOpInterface entryPoint,
+                                     IREE::LinalgExt::AttentionOp op) {
+  if (target.getWgp().getMma().empty())
+    return failure();
+
+  const int64_t targetSubgroupSize = target.getPreferredSubgroupSize();
+
+  // Get iteration domain bounds.
+  OpBuilder b(op);
+  SmallVector<Range> itDomain = op.getIterationDomain(b);
+  SmallVector<int64_t> bounds(itDomain.size());
+  for (auto [slice, bound] : llvm::zip_equal(itDomain, bounds)) {
+    OpFoldResult size = slice.size;
+    if (std::optional<int64_t> constSize = getConstantIntValue(size)) {
+      bound = constSize.value();
+    } else {
+      bound = ShapedType::kDynamic;
+    }
+  }
+
+  auto opInfo =
+      IREE::LinalgExt::AttentionOpDetail::get(op.getIndexingMapsArray())
+          .value();
+
+  int64_t mDim = opInfo.getMDims().back();
+  int64_t k1Dim = opInfo.getK1Dims().back();
+  int64_t k2Dim = opInfo.getK2Dims().back();
+  int64_t nDim = opInfo.getNDims().back();
+
+  // Dynamic dims are expected to be taken care of earlier in the pipeline.
+  if (ShapedType::isDynamic(bounds[mDim]) ||
+      ShapedType::isDynamic(bounds[k1Dim]) ||
+      ShapedType::isDynamic(bounds[k2Dim]) ||
+      ShapedType::isDynamic(bounds[nDim])) {
+    return failure();
+  }
+
+  // TODO: Do we need a matvec-like attention pipeline? Probably not,
+  // considering M is generally the largest dimension.
+
+  Value v = op.getValue();
```

Comment:
nit: more descriptive variable name

Reply:
done! (I think)

Reply:
Oh, now I see that `v` meant the `v` operand, not a shortening for `Value`. I guess it was fine before, but this is a little clearer, thanks!